dev-client/src/components/modals/PermissionsRequestModal.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useCallback, useRef} from 'react';
+import {useTranslation} from 'react-i18next';
+import {Linking} from 'react-native';
+
+import {createPermissionHook} from 'expo-modules-core';
+
+import {ConfirmModal} from 'terraso-mobile-client/components/modals/ConfirmModal';
+import {ModalHandle} from 'terraso-mobile-client/components/modals/Modal';
+
+type PermissionHook = ReturnType<typeof createPermissionHook>;
+
+type Props = {
+  title: string;
+  body: string;
+  usePermissions: PermissionHook;
+  permissionedAction?: () => void;
+  children: (onOpen: () => void) => React.ReactNode;
+};
+
+export const PermissionsRequestModal = ({
+  title,
+  body,
+  usePermissions,
+  permissionedAction,
+  children,
+}: Props) => {
+  const {t} = useTranslation();
+  const ref = useRef<ModalHandle>(null);
+  const [permissions, requestPermissions] = usePermissions();
+
+  const onRequestAction = useCallback(async () => {
+    if (permissions === null) {
+      return;
+    }
```

Comment:
question: can we check for any falsy value here, i.e. `!permissions`?

Reply:
the type of `permissions` is `PermissionsResponse | null` so it can't hold any other falsy value. if in the future it can hold a falsy value, we might want to handle it differently so i think it's good the type system will notify us that that's the case